internal/trust/remotes.go

```diff
@@ -111,7 +111,11 @@ func (r *Remotes) Add(dir string, remotes ...Remote) error {
 			return fmt.Errorf("Failed to parse remote %q to yaml: %w", remote.Name, err)
 		}
 
-		path := filepath.Join(dir, filepath.Base(remote.Name+".yaml"))
+		path, err := filepath.Abs(filepath.Join(dir, remote.Name+".yaml"))
+		if err != nil || !strings.HasPrefix(path, dir) {
+			return fmt.Errorf("Failed to get absolute path to %q yaml: %w", remote.Name, err)
```

Comment:
In case `err == nil` and `strings.HasPrefix(path, dir)` returns `false` the error message tries to include the `nil` error which is going to print something like `%!w(<nil>)`.

In the [example](https://github.com/canonical/microcluster/security/code-scanning/1) they just ignore the error but maybe it's best to have another `if` block after the call and check for `Abs` to transparently return the error (if any) and return something like "Invalid path %q to yaml" in case the prefix doesn't match.